### ISSUE TITLE
Empty title change rendering of siblings

### ISSUE THREAD

**Issue by alfsb** [php/phd#181]
I'm about to merge an [fixup](https://github.com/php/doc-base/pull/208) that may start including empty `<title>`s in manual, in an effort to translations can be build able for more time, particully at times that `doc-en` is being heavy updated.

The fixup `<title></title><simpara>%failed xi:xinclude%</simpara>` (empty title) is saved on `.manual.xml` as:
```xml
<refsect1 role="parameters">
 <title/>
 <simpara>%failed xi:xinclude%</simpara>
</refsect1
```
and is rendered by `php phd/render.php --format bigxhtml` as:

```html
<div class="refsect1 parameters" id="refsect1-function.bcdivmod-parameters">
  <h3 class="title"><p class="simpara">%failed xi:xinclude%</p></h3>
</div>
```

Instead of:

```html
<div class="refsect1 parameters" id="refsect1-function.bcdivmod-parameters">
  <h3 class="title"></h3>
  <p class="simpara">%failed xi:xinclude%</p>
</div>
```

In other words, empty `<title>` causes their siblings to be rendered inside the title, instead of around it. I'm worried that this can swallows entire sections inside titles. In the meanwhile, and to workaround this, I'm placing some placeholder texts in fixup titles, but these ends rendered in manual.

**Comment by haszi** [comment]
@alfsb I wrote a test for this in `PhD` (see [here](https://github.com/php/phd/commit/b08a625e8f0ebe7c80755577ff52a064705dec28)) but could not recreate the output you've got. 

I've tried `php phd/render.php --format bigxhtml`, `php phd/render.php --Generic --format bigxhtml` and `php phd/render.php --PHP --format bigxhtml` but neither produced that output. Could you share the xml markup somehow so I can try rendering that?

**Comment by alfsb** [comment]
> Could you share the xml markup somehow so I can try rendering that?

It's possible to generate it locally. Replace this line with:
```
$fixup = "<title></title><simpara>FAILEDXINCLUDE</simpara>";
```
Then clone `doc-it` as `it`, and `configure.php -q --with-lang=it`. After configure:
```
$ grep -B 2 -A 2 FAILEDXINCLUDE doc-base/.manual.xml 

 <refsect1 role="parameters">
  <title/><simpara>FAILEDXINCLUDE</simpara>
 </refsect1>

```
In the source, it does not get included inside title:
```
$ grep -B 2 -A 2 FAILEDXINCLUDE output/big-xhtml.html 

 <div class="refsect1 parameters" id="refsect1-function.bcdivmod-parameters">
  <h3 class="title"/><p class="simpara">FAILEDXINCLUDE</p>
 </div>

```

But it get rendered as the same as title:
![image](https://github.com/user-attachments/assets/bdff255d-4132-496c-b425-6b4a2ec411bc)

In DOM:
![image](https://github.com/user-attachments/assets/527c1c48-386d-4313-9575-826229743dce)

Probably because HTML `<H3>` are not expected to be autoclosed?

**Comment by alfsb** [comment]
Without changing `configure.php`, it would DOM and render ok:

![image](https://github.com/user-attachments/assets/1260b8e1-0e6b-46db-ba28-0557b6b5162e)

![image](https://github.com/user-attachments/assets/f99c1c8c-f47b-4324-b0fd-3d58be708408)

**Comment by haszi** [comment]
If I understand this correctly, the issue is that a valid self-closing XML/XHTML tag is not a valid HTML tag and as browsers render the PhD-generated pages as HTML, the browsers decide where to close these invalid HTML tags (probably before closing the self-closing tags' parent tags).

If that is so, the easiest fix for PhD would be to render the non-HTML self-closing tags as pairs of opening-closing tags.

**Comment by alfsb** [comment]
The fix may resides on saving the final HTML files with [`LIBXML_NOEMPTYTAG`](https://www.php.net/manual/en/libxml.constants.php#constant.libxml-noemptytag), or possibly better, saving HTML files with DOMDocument `saveHTML()`/`saveHTMLFile()`

**Comment by alfsb** [comment]
Also, this error is not detected by the test. As you said, this is basically a rendering issue on browsers, so it does not shows in saved files.
